src/nih_wayfinding/app/scripts/navigation/navigation-service.js

```diff
@@ -0,0 +1,112 @@
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function Navigation (
+        $q, $interval, $rootScope,
+        $geolocation,
+        NavigationStub
+    ) {
+
+        var stepsLeft = [];
+        var navInterval = 1000; // 1 second default interval
+        var module = {
+            walkTheLine: walkTheLine,
+            pollLocation: pollLocation,
+            getSteps: getSteps,
+            setInterval: setInterval,
+            stopIntervalTask: stopIntervalTask
+        };
+
+        return module;
+
+        /**
+         *  Change the interval for walking/checking geolocation
+         *
+         *  @param interval {number} The walk/read interval in milliseconds
+         *  @return undefined
+         */
+        function setInterval(interval) {
+            navInterval = interval;
+        }
+
+        /**
+         *  Return the steps that remain for walking the line
+         *
+         * @return {array} The steps that have been generated
+         *
+         */
+        function getSteps() {
+            return stepsLeft;
+        }
+
+        /**
+         * Stop the interval task (either checking GPS or iterating the walk of mocked data)
+         */
+        function stopIntervalTask() {
+            if (angular.isDefined(mockRun)) {
+                $interval.cancel(mockRun);
+                mockRun = undefined;
+            }
+            if (angular.isDefined(pollRun)) {
+                $interval.cancel(pollRun);
+                pollRun = undefined;
+            }
+        }
+
+        var mockRun;
+        /**
+         * Take the next step on the path as created by genSteps
+         *
+         * @param interval {number} Milliseconds of interval between steps
+         * @return {undefined}
+         */
+        function walkTheLine() {
+            var locations = NavigationStub;
+            stepsLeft = _.map(locations.features, function(feature) {
+                return feature.geometry.coordinates;
+            });
+            var runMock = function() {
+                mockRun = $interval(
+                    function() {
+                        if (stepsLeft.length > 0) {
+                            $rootScope.$broadcast('scanLoc', stepsLeft.shift());
+                        } else {
+                            stopIntervalTask();
+                        }
+                    },
+                    navInterval);
+            };
+            runMock();
+        }
+
+        var pollRun;
+        /**
+         * Poll for location using ngGeolocation module
+         *
+         * @return {undefined} Array representation of point: [[lng, lat], [lng, lat]]
+         */
+        function pollLocation() {
+            var runPoll = function() {
+                pollRun = $interval(
```

Comment:
Rather than using $interval and using getCurrentPosition, can't we just use the ngGeolocation.watchPosition function? It will be more efficient since the browser will only send update events when the position actually changes, rather than every x milliseconds. You can still inject $rootScope into this service and listen for events on it, forwarding them along.